drivers/pinctrl/bcm/pinctrl_bcm2835_rust.rs

```diff
@@ -0,0 +1,268 @@
+//SPDX-License-Identifier: GPL-2.0
+
+//! Driver for Boradcom BCM2835 GPIO unit (pinctrl + GPIO)
+//!
+//! Based on the C driver
+
+use core::result::Result::Ok;
+
+use kernel::{
+    bit, define_of_id_table, device, gpio,
+    io_mem::IoMem,
+    module_platform_driver, of, platform,
+    prelude::*,
+    sync::{Arc, ArcBorrow},
+};
+
+macro_rules! FSEL_REG {
+    ($p:expr) => {
+        GPFSEL0 + (($p / 10) * 4)
+    };
+}
+
+macro_rules! FSEL_SHIFT {
+    ($p:expr) => {
+        (($p % 10) * 3)
+    };
+}
+
+macro_rules! GPIO_REG_OFFSET {
+    ($p:expr) => {
+        $p / 32
+    };
+}
+
+macro_rules! GPIO_REG_SHIFT {
+    ($p:expr) => {
+        $p % 32
+    };
+}
+
+//GPIO register offsets
+const GPFSEL0: usize = 0x0; //function select
+const GPSET0: usize = 0x1c; //pin output set
+const GPCLR0: usize = 0x28; //pin output clear
+const GPLEV0: usize = 0x34; //pin level
+                            // const GPEDS0:usize = 0x40; //pin event detect Status
+                            // const GPREN0:usize = 0x4c; //pin rising edge detect enable
+                            // const GPFEN0:usize = 0x58; //pin falling edge detect enable
+                            // const GPHEN0:usize = 0x64; //pin high detect enable
+                            // const GPLEN0:usize = 0x70; //pin low detect enable
+                            // const GPAREN0:usize= 0x7c; //pin async rising edge detect
+                            // const GPAFEN0:usize= 0x88; // pin async falling edge detect
+                            // const GPPUD:usize = 0x94;  //pin pull-up/down enable
+                            // const GPPUDCLK0:usize = 0x98; // pin pull-up/down enable clock
+                            //TODO: no sure the precise offset size of BCM2835
```

Comment:
If you don't need these, just put them in your local repo. Don't send it in the pr.